impl/sql/mutationstorage/queue_test.go

```diff
@@ -19,25 +19,73 @@ import (
 	"testing"
 	"time"
 
+	"github.com/google/go-cmp/cmp"
 	"google.golang.org/grpc/codes"
 	"google.golang.org/grpc/status"
 
+	pb "github.com/google/keytransparency/core/api/v1/keytransparency_go_proto"
 	_ "github.com/mattn/go-sqlite3"
 )
 
+func TestRandShard(t *testing.T) {
+	ctx := context.Background()
+	db := newDB(t)
+	m, err := New(db)
+	if err != nil {
+		t.Fatalf("Failed to create Mutations: %v", err)
+	}
+
+	for _, tc := range []struct {
+		desc       string
+		send       bool
```

Comment:
Consider making this `send []int64` and trying some other cases (e.g. when len=1).